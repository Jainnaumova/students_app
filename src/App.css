.App {
  text-align: center;
  font-family: 'Roboto', sans-serif;

}

a {
  text-decoration: none;
  color:inherit;
}

.navbar-container {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 10vh;
  background-color: #006cb8;
  font-family: 'Serifa W01 45 Light',Times,Georgia,serif;
  font-size: 2rem;
  text-align: center;
}

.main-container {
  display: flex;
  min-height: 90vh;
}

.sidebar-container {
  font-family: 'Avenir LT',sans-serif;
  color: #797e8a;
  background-color: #e8ebec;
  width: 200px;
  box-shadow: 0 0.25rem 0.25rem rgba(0,0,0,.2), 0 0 1rem rgba(0,0,0,.2);
  padding-top: 0.7rem;
}

.selection-container {
  background-color: #fafcfd;
  width: 100%;
}

.selection-item {
  font-size: 1.2rem;
  padding-left: 0.8rem;
  margin-bottom: 0.8rem;
}

.items-group {
  display: flex;
  justify-content: space-around;
  margin-top: 5rem;
}

.item-button {
  display: flex;
  align-items: center;
  justify-content: center;
  color: #fff;
  background-color: #0072bb;
  border-radius: 3px;
  font-family: 'Serifa W01 45 Light',Times,Georgia,serif;
  font-size: 2rem;
  width: 14rem;
  height: 5rem;
  cursor: pointer;
}

.item-button:hover {
  background-color: #005c96;
}
