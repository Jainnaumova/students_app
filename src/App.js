import React from 'react';
import { HashRouter, Route, Switch } from 'react-router-dom';
import Navbar from './shared/Navbar';
import Home from './Home';
import Sidebar from './shared/Sidebar';
import Students from './students/Students';
import Teachers from './teachers/Teachers';

import './App.css';

const App = () => {
  return (
    <HashRouter>
      <div>
        <div className='navbar-container'>
          <Navbar />
        </div>
        <div className='main-container'>
          <div className='sidebar-container'>
            <Sidebar />
          </div>
          <div className='selection-container'>
            <Switch>
              <Route path="/" exact component={ Home } />
              <Route path="/students" exact component={ Students } />
              <Route path="/teachers" exact component={ Teachers } />
            </Switch>
          </div>
        </div>
      </div>
    </HashRouter>
  )
}

export default App;
