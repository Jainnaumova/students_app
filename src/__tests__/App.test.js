import React from 'react';
import ReactDOM from 'react-dom';
import Adapter from 'enzyme-adapter-react-15';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import App from '../App';

Enzyme.configure({ adapter: new Adapter() });

it('renders App component correctly', () => {
  const tree = renderer.create(
    <App />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
