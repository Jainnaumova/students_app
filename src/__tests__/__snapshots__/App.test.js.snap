// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component correctly 1`] = `
<div>
  <div
    className="navbar-container"
  >
    <div>
      New Visions for Public Schools
    </div>
  </div>
  <div
    className="main-container"
  >
    <div
      className="sidebar-container"
    >
      <div>
        <div
          className="selection-item"
        >
          <a
            href="#/"
            onClick={[Function]}
          >
            Home
          </a>
        </div>
        <div
          className="selection-item"
        >
          <a
            href="#/students"
            onClick={[Function]}
          >
            Students
          </a>
        </div>
        <div
          className="selection-item"
        >
          <a
            href="#/teachers"
            onClick={[Function]}
          >
            Teachers
          </a>
        </div>
      </div>
    </div>
    <div
      className="selection-container"
    >
      <div
        className="items-group"
      >
        <div
          className="item-button"
          onClick={[Function]}
        >
          Students
        </div>
        <div
          className="item-button"
          onClick={[Function]}
        >
          Teachers
        </div>
      </div>
    </div>
  </div>
</div>
`;
